invoice/src/main/java/org/killbill/billing/invoice/model/InvoiceItemCatalogBase.java

```diff
@@ -117,13 +115,21 @@ public String getPrettyUsageName() {
         return prettyUsageName;
     }
 
+    @Override
+    public Date getCatalogEffectiveDate() {
+        return catalogEffectiveDate;
+    }
+
     @Override
     public boolean matches(final Object o) {
 
         if (!super.matches(o)) {
             return false;
         }
         final InvoiceItemCatalogBase that = (InvoiceItemCatalogBase) o;
+        if (catalogEffectiveDate != null && that.catalogEffectiveDate != null ? catalogEffectiveDate.compareTo(that.catalogEffectiveDate) != 0 : that.catalogEffectiveDate != null) {
```

Comment:
⚠️ Not sure this is correct:

* `catalogEffectiveDate` = `now()`
* `that.catalogEffectiveDate` = `null`
* => `catalogEffectiveDate != null && that.catalogEffectiveDate` is `false` and `that.catalogEffectiveDate != null` if `false`
* => `if` is `false` but they are not equal?